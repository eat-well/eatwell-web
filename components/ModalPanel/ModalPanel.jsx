import { IconButton, Modal, Typography } from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import React from 'react';

const ModalPanel = ({ children, title, subtitle, open, onClose, classes, style }) => (
  <Modal open={open} onClose={onClose}>
    <div className={clsx(classes.modal, style)}>
      <div>
        <IconButton classes={{ root: classes.closeIcon }} onClick={onClose} aria-label='close'>
          <CloseIcon fontSize='small'/>
        </IconButton>
        <div className={classes.header}>
          <Typography variant='h6' noWrap={true} title={title}>{title}</Typography>
          <Typography variant='subtitle2' color='textSecondary'>
            {subtitle}
          </Typography>
        </div>
      </div>
      {children}
    </div>
  </Modal>
);

ModalPanel.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  open: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  title: PropTypes.string,
  style: PropTypes.string,
  subtitle: PropTypes.string,
};

export default ModalPanel;
