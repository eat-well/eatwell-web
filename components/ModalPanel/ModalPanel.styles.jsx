export default {
  closeIcon: {
    position: 'absolute',
    top: 10,
    right: 10,
  },
  modal: {
    margin: 'auto',
    backgroundColor: 'white',
    maxHeight: '80%',
    minHeight: 400,
    top: '10%',
    minWidth: 560,
    maxWidth: 800,
    borderRadius: 9,
    padding: '0 15px 10px 15px',
    '&:focus': {
      outline: 'none',
    },
    position: 'relative',
    display: 'flex',
    flexDirection: 'column',
  },
  header: {
    padding: '20px 40px',
    fontWeight: 'bold',
    maxWidth: 600,
    maxHeight: 500,
    margin: '0 auto',
    textAlign: 'center',
  },
  content: {
    maxHeight: 500,
    overflowY: 'auto',
    padding: '0 4px 4px',
    flex: 'auto',
  },
  modalFooter: {
    position: 'sticky',
    padding: 20,
    margin: '0 auto',
  }
};
