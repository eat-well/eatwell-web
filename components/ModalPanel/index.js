import { withStyles } from '@material-ui/core';
import styles from './ModalPanel.styles';
import ModalPanel from './ModalPanel';

export default withStyles(styles)(ModalPanel);
