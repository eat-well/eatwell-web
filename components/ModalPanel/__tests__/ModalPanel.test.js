import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import ModalPanel from 'components/ModalPanel';
import React from 'react';

const modalPanel = (
  <ModalPanel title='Modal title' open={true} onClose={jest.fn()} subtitle='Modal subtitle'>
    <span>This is a modal panel.</span>
  </ModalPanel>
);

describe('Modal Panel', () => {
  test('is rendered', () => {
    const {container} = render(modalPanel, { container: document.body });
    expect(container).toMatchSnapshot();
  });
});
