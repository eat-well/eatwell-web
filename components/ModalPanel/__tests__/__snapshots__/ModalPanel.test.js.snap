// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Panel is rendered 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      class="ModalPanel-modal-2"
      tabindex="-1"
    >
      <div>
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root ModalPanel-closeIcon-1"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          class="ModalPanel-header-3"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
          >
            Modal title
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextSecondary"
          >
            Modal subtitle
          </h6>
        </div>
      </div>
      <span>
        This is a modal panel.
      </span>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
